### PR TITLE
Fix PageStore FindBugs issues

### PR DESCRIPTION
**Pull request by bf8086**
Fixed following issues reported by findbugs:

```
[ERROR] Possible null pointer dereference in alluxio.client.file.cache.store.LocalPageStore.delete(PageId) due to return value of called method [alluxio.client.file.cache.store.LocalPageStore, alluxio.client.file.cache.store.LocalPageStore] Method invoked at LocalPageStore.java:[line 89]Known null at LocalPageStore.java:[line 89] NP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE
[ERROR] Possible null pointer dereference in alluxio.client.file.cache.store.LocalPageStore.delete(PageId) due to return value of called method [alluxio.client.file.cache.store.LocalPageStore, alluxio.client.file.cache.store.LocalPageStore] Method invoked at LocalPageStore.java:[line 90]Known null at LocalPageStore.java:[line 90] NP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE
[ERROR] Possible null pointer dereference in alluxio.client.file.cache.store.LocalPageStore.put(PageId, byte[]) due to return value of called method [alluxio.client.file.cache.store.LocalPageStore, alluxio.client.file.cache.store.LocalPageStore] Method invoked at LocalPageStore.java:[line 59]Known null at LocalPageStore.java:[line 59] NP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE
[ERROR] alluxio.client.file.cache.store.LocalPageStore.delete(PageId) may fail to close stream [alluxio.client.file.cache.store.LocalPageStore] At LocalPageStore.java:[line 89] OS_OPEN_STREAM
[ERROR] Unused field: alluxio.client.file.cache.store.LocalPageStoreOptions.mRootDir [alluxio.client.file.cache.store.LocalPageStoreOptions] In LocalPageStoreOptions.java UUF_UNUSED_FIELD
```

### CODE REVIEW

**Comment by bf8086** [comment]
@apc999 @calvinjia PTAL.

File: core/client/fs/src/main/java/alluxio/client/file/cache/store/LocalPageStore.java [hunk 1]
```diff
@@ -55,7 +57,7 @@ public LocalPageStore(LocalPageStoreOptions options) {
   public void put(PageId pageId, byte[] page) throws IOException {
     Path p = getFilePath(pageId);
     if (!Files.exists(p)) {
-      Files.createDirectories(p.getParent());
+      Files.createDirectories(Preconditions.checkNotNull(p.getParent(), "parent"));
```
Comment (apc999) [inline core/client/fs/src/main/java/alluxio/client/file/cache/store/LocalPageStore.java]: nit: for src code readability & more explicit error message
```
Preconditions.checkNotNull(p.getParent(), "Path for page %s can not be root", pageId);
Files.createDirectories(p.getParent());
```

File: core/client/fs/src/main/java/alluxio/client/file/cache/store/LocalPageStore.java [hunk 2]
```diff
@@ -82,8 +84,11 @@ public void delete(PageId pageId) throws IOException, PageNotFoundException {
     }
     Files.delete(p);
     mSize.decrementAndGet();
-    if (!Files.newDirectoryStream(p.getParent()).iterator().hasNext()) {
-      Files.delete(p.getParent());
+    Path parent = Preconditions.checkNotNull(p.getParent(), "parent");
```
Comment (apc999) [inline core/client/fs/src/main/java/alluxio/client/file/cache/store/LocalPageStore.java]: please make the error message more explicit.

File: core/client/fs/src/main/java/alluxio/client/file/cache/store/LocalPageStore.java (hunk 1 above)
Comment (bf8086) [reply core/client/fs/src/main/java/alluxio/client/file/cache/store/LocalPageStore.java]: Done.

File: core/client/fs/src/main/java/alluxio/client/file/cache/store/LocalPageStore.java (hunk 2 above)
Comment (bf8086) [reply core/client/fs/src/main/java/alluxio/client/file/cache/store/LocalPageStore.java]: Done.

**Comment by apc999** [comment]
alluxio-bot, merge this please

**Pull request merged**